### DISCUSSION TITLE
mengambil elemens agar tidak tercampur

### DISCUSSION THREAD

**Discussion by Imam-Danu** [teamalgoritma/community#45] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/101544652/162675935-c622dfd0-e1c5-41be-912c-ab8d7bf94d0f.png)

bagaimana caranya mengambil salah satu bagiannya saja?
table.find_all('td', attrs={'class':'text-center'})[i].text
kalau saya ketik syntax tersebut pasti akan tercampur
namun bila syntax: table.find_all('td', attrs={'class':'text-center'})[1].text
semua data berisi data yg sama
solusinya?

**Comment by fafilia** [comment, marked as answer]
Anda bisa menggunakan fungsi perulangan untuk mengambil nilai spesifik dari komponen tertentu untuk seluruh baris. Contoh misal dari data tersebut kita hanya akan mengambil nilai ke-2. Untuk membuat fungsi perulangan, yang pertama perlu dilakukan adalah mengetahui berapa maksimal dari baris yang akan diambil datanya. Dalam hal ini Anda dapat menggunakan length(). Sedangkan untuk perulangannya sendiri Anda dapat mendefinisikannya menggunakan perintah for seperti contoh berikut:


```
temp =[] #menyimpan value yang discrap

for index in range(1, row_length): 
     data= row[index].find_all('td')[1].text #mengambil nilai tertentu dari index tertentu
     temp.append(data) # menggabungkan nilai scrap masing-masing baris

temp # return value
```

**Reply by Imam-Danu** [reply]
masih error terus kak...

**Comment by Hertzspr** [comment]
Bisa dicoba dengan nyari pola. 
**MISAL**, yang mau dicari di kolom ke 3 (index [2])

0       1     [2]     3
4       5     [6]     7
8       9   [10]   11
12   13   [14]   15

yg muncul:
putaran pertama index 0 keluar index 2
putaran kedua 1 keluar index 6
putaran ketuga index 2 keluar index 10
putaran keempat index 3 keluar index 14

kita lihat kalau i = 0 di putaran pertama, sementara yang mau dicari ada di [2]
gimana kira kira hubungan matematiknya? 
karena terdiri dari 4 kolom, kita coba kali empat,
karena yg dicari misalnya ada di kolom ketiga, kita coba tambah 3,
kalau yang mau dicari kita sebut sebagai j, kita buat persamaan

j = i * 4 + 3, 
kalau i = 0, maka j 
j = 0 * 4 + 3 = 3 
karena j = 3, maka nanti di loop dia ngambil index [2]
0 - 1 - [2] - 3 


kalau i = 1 di putaran kedua, sementara yang mau dicari ada di [6]
gimana kira=kira?
j = i * 4 + 3
kalau i = , maka j
j = 1 * 4 + 3 = 7
karena j = 7, maka nanti di loop dia ngambi index [6]

Begitu kira-kira logikanya, untuk kemudian diterapkan ke loop yang mau dibuat.
